src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -82,6 +87,13 @@
      */
     boolean isCreated();
 
+    /**
+     * gets model index's health status
+     *
+     * @return ClusterHealthStatus of model index
+     */
+    ClusterHealthStatus getHealthStatus();
```

Comment:
Why is this cluster health status when we are interested in index status?

Reply:
Unfortunately, this Enum is only being used to represent IndexHealth status as well since value is common among both index and cluster.